File: sources/CPP/include/Ers/Visualization/Material.cpp
#include "Material.h"
#include "Ers/Api.h"

namespace Ers
{
    Material::Material(void* corePointer) :
        data(corePointer)
    {
    }

    Ers::Color Material::GetColor() const
    {
        return Ers::Color::FromInt(Ers::Engine::ERS_Material_GetDiffuseColor(data));
    }

    void Material::SetColor(Ers::Color color)
    {
        Ers::Engine::ERS_Material_SetDiffuseColor(data, color.Value);
    }

    void* Material::CorePtr()
    {
        return data;
    }

    const void* const Material::CorePtr() const
    {
        return data;
    }
} // namespace Ers

File: sources/CSharp/src/Ers/Visualization/Material.cs
using Ers.Engine;

namespace Ers
{
    /// <summary>
    /// Material containing color and texture information.
    /// </summary>
    public class Material
    {
        /// <summary>
        /// Native pointer to the core instance.
        /// </summary>
        public readonly IntPtr CorePtr;

        internal Material(IntPtr corePtr) { CorePtr = corePtr; }

        /// <summary>
        /// The diffuse color of the material.
        /// </summary>
        public Color Color
        {
            get => Color.FromInt(ErsEngine.ERS_Material_GetDiffuseColor(CorePtr));
            set => ErsEngine.ERS_Material_SetDiffuseColor(CorePtr, value.Value);
        }
    }
}
